examples/cosmos/lightclient/Lightclient.qnt

```diff
@@ -0,0 +1,550 @@
+    // the assumed precision of the clock
+    const CLOCK_DRIFT: int
+    // the actual clock drift, which under normal circumstances should not
+    // be larger than CLOCK_DRIFT (otherwise, there will be a bug)
+    const REAL_CLOCK_DRIFT: int
+    // Flag taht determines whether primary is correct. Modifies the behavior of FetchLightBlockInto, and invariants.
+    const IS_PRIMARY_CORRECT: bool
+    // a pair (a, b) that limits that ratio of faulty validator in the blockchain
+    // from above (exclusive). Tendermint security model prescribes 1 / 3.
+    const FAULTY_RATIO: (int, int)
+
+    // the current state of the light client
+    var state: str
+    // the next height to explore by the light client
+    var nextHeight: int
+    // the light client iteration, or the number of block tests
+    var nprobes: int
+  
+    /* the light store */
+
+    // a function from heights to LightBlocks
+    var fetchedLightBlocks: LightBlockMap
+    // a function from heights to block statuses
+    var lightBlockStatus: LightBlockStatus
+    // the latest verified block
+    var latestVerified: LightBlock
+
+    /* the light client previous state components, used for monitoring */
+    var prevVerified: LightBlock
+    var prevCurrent: LightBlock
+    var prevLocalClock: int 
+    var prevVerdict: str
+
+
+    // Unlike the TLA file, we can use the same syntax for InitMonitor and NextMonitor
+    action MonitorTransition(verified: LightBlock, current: LightBlock, pLocalClock: int, verdict: str): bool = all {
+        prevVerified' = verified,
+        prevCurrent' = current,
+        prevLocalClock' = pLocalClock,
+        prevVerdict' = verdict
+    }
+
+    /******************* Blockchain instance ***********************************/
+
+    // A set of all nodes that can act as validators (correct and faulty)
+    const AllNodes: Set[str]
+
+    pure def ULTIMATE_HEIGHT = TARGET_HEIGHT + 1 
+
+    /* Create an instance of Blockchain. */ 
+    import Blockchain(
+        AllNodes = AllNodes, 
+        ULTIMATE_HEIGHT = ULTIMATE_HEIGHT, 
+        TRUSTING_PERIOD = TRUSTING_PERIOD
+        ) as BC from "./Blockchain"
+
+    /************************** Lite client ************************************/
+
+    /* the heights on which the light client is working */  
+    pure def HEIGHTS = TRUSTED_HEIGHT.to(TARGET_HEIGHT)
+
+    /* the control states of the lite client */ 
+    pure def States = Set( "working", "finishedSuccess", "finishedFailure" )
+
+    // The verification functions are implemented in the API
+    import LCVerificationApi(
+        TRUSTING_PERIOD = TRUSTING_PERIOD, 
+        CLOCK_DRIFT = CLOCK_DRIFT,
+        REAL_CLOCK_DRIFT = REAL_CLOCK_DRIFT,
+        FAULTY_RATIO = FAULTY_RATIO
+        ) as API from "./LCVerificationApi"
+
+
+    /*
+    Initial states of the light client.
+    Initially, only the trusted light block is present.
+    */
+    action LCInit: bool = all {
+        {
+            nondet tm = oneOf(Int)
```

Comment:
Let's use `Nat` instead of `Int`. It would not make a difference in Apalache, but the Quint simulator would choose negative integers about half of the time.